openfeature/client_test.go

```diff
@@ -1112,3 +1113,195 @@ func TestFlagMetadataAccessors(t *testing.T) {
+	}, time.Second, 100*time.Millisecond, "expected client to report NOT READY state")
+
+	provider := struct {
+		FeatureProvider
+		EventHandler
+	}{
+		NoopProvider{},
+		&ProviderEventing{
+			c: make(chan Event, 1),
+		},
+	}
+
+	if err := SetProviderAndWait(provider); err != nil {
+		t.Fatalf("failed to set up provider: %v", err)
+	}
+	eventually(t, func() bool {
+		return GetApiInstance().GetClient().State() == ReadyState
+	}, time.Second, 100*time.Millisecond, "expected client to report READY state")
+}
+
+// The client's provider status accessor MUST indicate ERROR if the initialize function of the associated provider
+// terminates abnormally
+func TestRequirement_1_7_3(t *testing.T) {
+	defer t.Cleanup(initSingleton)
+	provider := struct {
+		FeatureProvider
+		StateHandler
+		EventHandler
+	}{
+		NoopProvider{},
+		&stateHandlerForTests{
+			initF: func(e EvaluationContext) error {
+				return errors.New("whoops... error from initialization")
+			},
+		},
+		&ProviderEventing{},
+	}
+
+	_ = SetProviderAndWait(provider)
+	eventually(t, func() bool {
+		return GetApiInstance().GetClient().State() == ErrorState
+	}, time.Second, 100*time.Millisecond, "expected client to report ERROR state")
+}
+
+// The client's provider status accessor MUST indicate FATAL if the initialize function of the associated provider
+// terminates abnormally and indicates error code PROVIDER_FATAL.
+func TestRequirement_1_7_4(t *testing.T) {
+	defer t.Cleanup(initSingleton)
+	provider := struct {
+		FeatureProvider
+		StateHandler
+		EventHandler
+	}{
+		NoopProvider{},
+		&stateHandlerForTests{
+			initF: func(e EvaluationContext) error {
+				return errors.New("whoops... error from initialization")
+			},
+		},
+		&ProviderEventing{},
+	}
+
+	_ = SetProviderAndWait(provider)
+
+	eventually(t, func() bool {
+		return GetApiInstance().GetClient().State() == ErrorState
+	}, time.Second, 100*time.Millisecond, "expected client to report ERROR state")
+}
+
+// The client's provider status accessor MUST indicate FATAL if the initialize function of the associated provider
+// terminates abnormally and indicates error code PROVIDER_FATAL.
+func TestRequirement_1_7_5(t *testing.T) {
+	defer t.Cleanup(initSingleton)
+	provider := struct {
+		FeatureProvider
+		StateHandler
+		EventHandler
+	}{
+		NoopProvider{},
+		&stateHandlerForTests{
+			initF: func(e EvaluationContext) error {
+				return &ProviderInitError{ErrorCode: ProviderFatalCode}
+			},
+		},
+		&ProviderEventing{},
+	}
+
+	_ = SetProviderAndWait(provider)
+
+	eventually(t, func() bool {
+		return GetApiInstance().GetClient().State() == FatalState
+	}, time.Second, 100*time.Millisecond, "expected client to report ERROR state")
+}
+
+// The client MUST default, run error hooks, and indicate an error if flag resolution is attempted while the provider
+// is in NOT_READY.
+func TestRequirement_1_7_6(t *testing.T) {
+	defer t.Cleanup(initSingleton)
+	t.Skip("Test not yet implemented")
```

Comment:
Did you mean to implement these? (1.7.6 - 1.7.8)?

It seems to me that the short-circuiting logic is missing for NOT_READY and FATAL, at least. I think you'd need to implement that in `client.evaluate`.

Reply:
yes, let's do that in a another PR

Reply:
I think you can complete these now, yes? Or would you rather leave them to another PR?